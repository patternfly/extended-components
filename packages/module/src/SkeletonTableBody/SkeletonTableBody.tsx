import React from 'react';
import {
  Tbody,
  Td,
  Tr,
  RowSelectVariant
} from '@patternfly/react-table';
import { Skeleton } from '@patternfly/react-core';

export interface SkeletonTableBodyProps {
  /** Number of columns in the table */
  columnsCount: number;
  /** Number of rows in the table */
  rowsCount?: number;
  /** Custom OUIA ID */
  ouiaId?: string | number;
  /** Flag indicating if the table is selectable */
  isSelectable?: boolean;
  /** Flag indicating if the table is expandable */
  isExpandable?: boolean;
  /** Determines if the row selection variant (radio/checkbox) */
  selectVariant?: RowSelectVariant;
}

const SkeletonTableBody: React.FunctionComponent<SkeletonTableBodyProps> = ({
  rowsCount = 5,
  ouiaId = 'SkeletonTableBody',
  isSelectable,
  isExpandable,
  selectVariant = RowSelectVariant.checkbox,
  columnsCount,
}: SkeletonTableBodyProps) => {
  const bodyRows = [ ...Array(rowsCount) ].map((_, rowIndex) => (
    <Tr key={rowIndex} ouiaId={`${ouiaId}-tr-${rowIndex}`}>
      {isExpandable && (
        <Td
          data-ouia-component-id={`${ouiaId}-td-expand-${rowIndex}`}
          expand={{
            rowIndex,
            isExpanded: false
          }}
        />
      )}
      {isSelectable && (
        <Td
          data-ouia-component-id={`${ouiaId}-td-select-${rowIndex}`}
          select={{
            rowIndex,
            isSelected: false,
            isDisabled: true,
            variant: selectVariant
          }}
        />
      )}
      {[ ...Array(columnsCount) ].map((_, colIndex) => (
        <Td key={colIndex} data-ouia-component-id={`${ouiaId}-td-${rowIndex}-${colIndex}`}>
          <Skeleton />
        </Td>
      ))}
    </Tr>
  ));

  return (
    <Tbody data-ouia-component-id={`${ouiaId}-tbody`}>{bodyRows}</Tbody>
  );
};

export default SkeletonTableBody;
