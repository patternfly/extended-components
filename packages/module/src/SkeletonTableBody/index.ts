export { default } from './SkeletonTableBody';
export * from './SkeletonTableBody';
