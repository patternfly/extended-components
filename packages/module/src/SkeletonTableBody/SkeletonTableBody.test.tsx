import React from 'react';
import { render } from '@testing-library/react';
import SkeletonTableBody from './SkeletonTableBody';

describe('SkeletonTableBody component', () => {
  it('should render correctly', () => {
    expect(render(<SkeletonTableBody columnsCount={2} />)).toMatchSnapshot();
  });
});
