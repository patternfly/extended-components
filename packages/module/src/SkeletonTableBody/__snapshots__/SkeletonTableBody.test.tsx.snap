// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonTableBody component should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tbody
        class="pf-v6-c-table__tbody"
        data-ouia-component-id="SkeletonTableBody-tbody"
        role="rowgroup"
      >
        <tr
          class="pf-v6-c-table__tr"
          data-ouia-component-id="SkeletonTableBody-tr-0"
          data-ouia-component-type="PF6/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTableBody-td-0-0"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTableBody-td-0-1"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v6-c-table__tr"
          data-ouia-component-id="SkeletonTableBody-tr-1"
          data-ouia-component-type="PF6/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTableBody-td-1-0"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTableBody-td-1-1"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v6-c-table__tr"
          data-ouia-component-id="SkeletonTableBody-tr-2"
          data-ouia-component-type="PF6/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTableBody-td-2-0"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTableBody-td-2-1"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v6-c-table__tr"
          data-ouia-component-id="SkeletonTableBody-tr-3"
          data-ouia-component-type="PF6/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTableBody-td-3-0"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTableBody-td-3-1"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v6-c-table__tr"
          data-ouia-component-id="SkeletonTableBody-tr-4"
          data-ouia-component-type="PF6/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTableBody-td-4-0"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v6-c-table__td"
            data-ouia-component-id="SkeletonTableBody-td-4-1"
            tabindex="-1"
          >
            <div
              class="pf-v6-c-skeleton"
            >
              <span
                class="pf-v6-screen-reader"
              />
            </div>
          </td>
        </tr>
      </tbody>
    </div>
  </body>,
  "container": <div>
    <tbody
      class="pf-v6-c-table__tbody"
      data-ouia-component-id="SkeletonTableBody-tbody"
      role="rowgroup"
    >
      <tr
        class="pf-v6-c-table__tr"
        data-ouia-component-id="SkeletonTableBody-tr-0"
        data-ouia-component-type="PF6/TableRow"
        data-ouia-safe="true"
      >
        <td
          class="pf-v6-c-table__td"
          data-ouia-component-id="SkeletonTableBody-td-0-0"
          tabindex="-1"
        >
          <div
            class="pf-v6-c-skeleton"
          >
            <span
              class="pf-v6-screen-reader"
            />
          </div>
        </td>
        <td
          class="pf-v6-c-table__td"
          data-ouia-component-id="SkeletonTableBody-td-0-1"
          tabindex="-1"
        >
          <div
            class="pf-v6-c-skeleton"
          >
            <span
              class="pf-v6-screen-reader"
            />
          </div>
        </td>
      </tr>
      <tr
        class="pf-v6-c-table__tr"
        data-ouia-component-id="SkeletonTableBody-tr-1"
        data-ouia-component-type="PF6/TableRow"
        data-ouia-safe="true"
      >
        <td
          class="pf-v6-c-table__td"
          data-ouia-component-id="SkeletonTableBody-td-1-0"
          tabindex="-1"
        >
          <div
            class="pf-v6-c-skeleton"
          >
            <span
              class="pf-v6-screen-reader"
            />
          </div>
        </td>
        <td
          class="pf-v6-c-table__td"
          data-ouia-component-id="SkeletonTableBody-td-1-1"
          tabindex="-1"
        >
          <div
            class="pf-v6-c-skeleton"
          >
            <span
              class="pf-v6-screen-reader"
            />
          </div>
        </td>
      </tr>
      <tr
        class="pf-v6-c-table__tr"
        data-ouia-component-id="SkeletonTableBody-tr-2"
        data-ouia-component-type="PF6/TableRow"
        data-ouia-safe="true"
      >
        <td
          class="pf-v6-c-table__td"
          data-ouia-component-id="SkeletonTableBody-td-2-0"
          tabindex="-1"
        >
          <div
            class="pf-v6-c-skeleton"
          >
            <span
              class="pf-v6-screen-reader"
            />
          </div>
        </td>
        <td
          class="pf-v6-c-table__td"
          data-ouia-component-id="SkeletonTableBody-td-2-1"
          tabindex="-1"
        >
          <div
            class="pf-v6-c-skeleton"
          >
            <span
              class="pf-v6-screen-reader"
            />
          </div>
        </td>
      </tr>
      <tr
        class="pf-v6-c-table__tr"
        data-ouia-component-id="SkeletonTableBody-tr-3"
        data-ouia-component-type="PF6/TableRow"
        data-ouia-safe="true"
      >
        <td
          class="pf-v6-c-table__td"
          data-ouia-component-id="SkeletonTableBody-td-3-0"
          tabindex="-1"
        >
          <div
            class="pf-v6-c-skeleton"
          >
            <span
              class="pf-v6-screen-reader"
            />
          </div>
        </td>
        <td
          class="pf-v6-c-table__td"
          data-ouia-component-id="SkeletonTableBody-td-3-1"
          tabindex="-1"
        >
          <div
            class="pf-v6-c-skeleton"
          >
            <span
              class="pf-v6-screen-reader"
            />
          </div>
        </td>
      </tr>
      <tr
        class="pf-v6-c-table__tr"
        data-ouia-component-id="SkeletonTableBody-tr-4"
        data-ouia-component-type="PF6/TableRow"
        data-ouia-safe="true"
      >
        <td
          class="pf-v6-c-table__td"
          data-ouia-component-id="SkeletonTableBody-td-4-0"
          tabindex="-1"
        >
          <div
            class="pf-v6-c-skeleton"
          >
            <span
              class="pf-v6-screen-reader"
            />
          </div>
        </td>
        <td
          class="pf-v6-c-table__td"
          data-ouia-component-id="SkeletonTableBody-td-4-1"
          tabindex="-1"
        >
          <div
            class="pf-v6-c-skeleton"
          >
            <span
              class="pf-v6-screen-reader"
            />
          </div>
        </td>
      </tr>
    </tbody>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
